### PR TITLE
Large library fixes

### PR DESCRIPTION
**Pull request by rafntor**
Hopefully fixes for #108 and #97, at least my collection can now be set up and scan started
Python novice, so please inspect carefully and advice if something looks bad.
Several independent commits for different issues

### CODE REVIEW

File: api/directory_watcher.py [hunk 1]
```diff
@@ -29,15 +29,19 @@ def calculate_hash(user,image_path):
     return hash_md5.hexdigest() + str(user.id)
 import os
```
Comment (derneuere) [inline api/directory_watcher.py:30]: Not sure how that import os got there. I don't think we need it because at the start of the file

**Comment by derneuere** [comment]
Other than that I found a couple of formatting inconsistencies. We use black as formatter for python :+1:

**Comment by derneuere** [comment]
Alright. My scan was successful. I will merge the changes and fix the small things above. Thanks again for your pull request :+1:

**Comment by rafntor** [comment]
Great ! Please notice the recursion limitation in `path_to_dict` is just a workaround and maybe need to be addressed again in the future in the way you already suggested with on-demand-lookup when user clicks the tree

File: api/directory_watcher.py (hunk 1 above)
Comment (rafntor) [reply api/directory_watcher.py:30]: Dont know either, I dont see it in the files changed(?)

**Pull request merged**